tests/steamship_tests/utils/random.py

```diff
@@ -19,9 +19,11 @@ def random_name() -> str:
 
 @contextlib.contextmanager
 def random_index(steamship: Steamship, plugin_instance: str) -> EmbeddingIndex:
-    index = EmbeddingIndex.create(
-        client=steamship,
-        plugin_instance=plugin_instance,
+    index = steamship.use_plugin(
+        "embedding-index",
+        random_name(),
+        config={"embedder": {"plugin_handle": plugin_instance, "fetch_if_exists": True}},
```

Comment:
Does it make sense to ask the user to instantiate the embedder themselves and pass the plugin_instance_handle of the embedder? 

That would result in 2 use_plugin calls but would reduce the interface of the config object and potentially simplify the logic of the embedding-index.

Reply:
If you're comfortable with it -- I'm going to push forward on this and return to this comment in a future PR. 

I think both have some advantages, and it'll be easy to experiment with that in a way we can all take a look at a PRs downstream from this.